File: DataStructures/LinkedList/DeleteLinkedListFunction.cs
using System;

public class LinkedList
{
    Node head;

    class Node
    {
        public int data;
        public Node next;
        public Node (int d)
        {
            data = d;
            next = null;
        }

    }

    void deleteList()
    {
        Node tnode = head;
        Console.Write("List to be deleted: ");
        while(tnode != null)
        {
            Console.Write(tnode.data + " ");
            tnode = tnode.next;
        }
        head = null;
    }

    void push(int new_data)
    {
        Node new_node = new Node(new_data);
        new_node.next = head;
        head = new_node;
    }


    public static void Main(String[] args)
    {
       LinkedList llist = new LinkedList();

       llist.push(1);
       llist.push(2);
       llist.push(3);
       llist.push(14);
       llist.push(15;

       llist.deleteList();

    }


}

File: DataStructures/LinkedList/DeleteLinkedListFunction.py
class Node:
    def __init__(self,data):
        self.data = data
        self.next = None

class LinkedList:

    def__init(self):
    self.head = None

    def deleteList(self):
        current = self.head

        while current:
            next = current.next

            del current.data

            current = next

    def push(self, new_data):
        new_node = Node(new_data)

        new_node.next = self.head

        self.head = new_node

if __name__ == '__main__':

    llist = LinkedList()
    llist.push(1)
    llist.push(6)
    llist.push(5)
    llist.push(4)
    llist.push(3)

    print("Deleting linked list")
    llist.deleteList()
    print("Linked list deleted")